src/python/pants/backend/python/dependency_inference/scripts/import_parser.py

```diff
@@ -100,40 +105,23 @@ def visit_Call(self, node):
             if name is not None:
                 lineno = node.args[0].lineno
                 if not self._is_pragma_ignored(self._contents_lines[lineno - 1]):
-                    self.imports.setdefault(name, lineno)
+                    self.definite_imports.setdefault(name, lineno)
                 return
 
         self.generic_visit(node)
 
+    # For Python 2.7, and Python3 < 3.8
+    def visit_Str(self, node):
+        try:
+            val = node.s.decode("utf8") if isinstance(node.s, bytes) else node.s
+            self.maybe_add_string_import(node, val)
+        except UnicodeError:
+            pass
 
-if os.environ["STRING_IMPORTS"] == "y":
-    # String handling changes a bit depending on Python version. We dynamically add the appropriate
-    # logic.
-    if sys.version_info[0:2] == (2, 7):
-
-        def visit_Str(self, node):
-            try:
-                val = node.s.decode("utf8") if isinstance(node.s, bytes) else node.s
-                self.maybe_add_string_import(node, val)
-            except UnicodeError:
-                pass
-
-        setattr(AstVisitor, "visit_Str", visit_Str)
-
-    elif sys.version_info[0:2] < (3, 8):
-
-        def visit_Str(self, node):
-            self.maybe_add_string_import(node, node.s)
-
-        setattr(AstVisitor, "visit_Str", visit_Str)
-
-    else:
-
-        def visit_Constant(self, node):
-            if isinstance(node.value, str):
-                self.maybe_add_string_import(node, node.value)
-
-        setattr(AstVisitor, "visit_Constant", visit_Constant)
```

Comment:
This isn't necessary for this change, but was a cleanup/refactor I've been meaning to do. The new way is simpler and (as the tests show) still work :tada:

Reply:
The old way was intentional. We don't want to handle bytes with Python 3. But actually I guess that wasn't necessary? `Str` and `Bytes` are different in Python 3's AST: https://docs.python.org/3.7/library/ast.html#abstract-grammar. So yeah looks good :)